.github/workflows/test.yml

```diff
@@ -19,28 +19,39 @@ jobs:
       fail-fast: false
       matrix:
         env: ${{ fromJSON(needs.getcidata.outputs.environments) }}
-        include:
-          - services: {}
-            redis_host: localhost
-          - env: { os: ubuntu-latest }
-            services: { redis: { image: redis, ports: [ '6379:6379' ] } }
-            redis_host: redis
     runs-on: ${{ matrix.env.os }}
     container: ${{ matrix.env.image }}
-    services: ${{ matrix.services }}
     steps: 
       - name: Select toolchain
         uses: maxim-lobanov/setup-xcode@v1.2.1
         with:
           xcode-version: ${{ matrix.env.toolchain }}
         if: ${{ matrix.env.toolchain != '' }}
-      - name: Set up Redis server via Homebrew
-        if: ${{ matrix.env.toolchain != '' }}
-        run: brew install redis && brew services start redis
-      - name: Check out Redis
+      - name: Check out source code
         uses: actions/checkout@v2
-      - name: Run tests with Thread Sanitizer
+
+      - name: Install Redis via Homebrew (Mac)
+        if: ${{ matrix.env.toolchain != '' }}
+        run: brew install redis 
+
+      - name: Install Redis via apt-get (Ubuntu)
+        if: ${{ matrix.env.toolchain == '' }}
+        run: apt-get update && apt-get install -y redis
+
```

Comment:
Add this:
```yaml
      - name: Install Redis via ??? (Windows)
         if: ${{ startsWith(matrix.env.os, 'windows') }}
         run: echo NOT IMPLEMENTED; exit 1
```
So there's somewhere to put a Windows equivalent in the future 🙂.

(This assuming I don't eventually use composite run actions etc. to provide this kind of thing automagically.)